### ISSUE TITLE
Errors package

### ISSUE THREAD

**Issue by blokovi** [absmach/magistrala#870]
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.
No

2. Describe the feature you are requesting, as well as the possible use case(s) for it.
In order to better handle errors and logs (issue #538 ) we could use package that will be wrapper around Go 1.13 stdlib error package. This package should be part of shared utilities as described in #666 so it could be used in all Mainflux services
Use cases:
- `errors.Unwrap` returns error that is wrapped, but we could use function that would return wrapper itself (wrapper - string that is error wrapped with). We need this wrapper for displaying in API responses - just nice string that last error is wrapped with.
- we need wrapping (combining) two errors. When we get error from i.e. adapter and want that error to wrap with our custome error and return. We need both errors to be combined, so either of the two can be recognized with `error.Is`

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).
Nice to have

**Comment by drasko** [comment]
@blokovi isn't this a duplicate of https://github.com/mainflux/mainflux/issues/538?

**Comment by anovakovic01** [comment]
@drasko #538 is too large to manage. We'll have multiple PRs in order to solve this, and this issue only addresses the shared `errors` API issue that @blokovi described.

**Comment by blokovi** [comment]
After getting into details what I propose is this:
We can use error wrapping from standard Go 1.13 lib like this:
```
ErrNotFound = errors.New("non-existent entity")
if err == sql.ErrNoRows {
  return fmt.Errorf("%s: %w", ErrNotFound, err)
}
```
When error is created in our code (not forwarded from other package) then :
```
return fmt.Errorf("Not found message: %w", ErrNotFound)
```

We will create function that can return wrapper of wrapped error (unlike `errors.Unwrap` that returns error that is wrapped):
```
func Wrapper(err error) string {
	wrapper := strings.SplitN(err.Error(), ":", 2)
	return wrapper[0]
}
```
We use this for displaying in API responses as described in first message in this thread. Also we are using it in next function that our package must have:

`func Has(err, target error) bool`
In error checking (ie transport.go) we will use `Has` function  (similiar to `error.Is`) that compares string of target error that `Error()` returns with wrapper of err. If it finds it in any wrapper or last wrapped error it returns `true`. That is how we can check if target error is in wrapped chain. 
In encodeError we can use it similiar to this:
```
switch {
case Has(err, ErrNotFound):
  w.WriteHeader("404")
```

**Comment by nmarcetic** [comment]
Closed with #146
